### PR TITLE
added rule documentation

### PR DESCRIPTION
**Pull request by kaikreuzer**
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
@rkoshak, @watou: Any volunteer to review this? Note that most content of the textual rule docs are from the OH1 wiki.

**Comment by kaikreuzer** [comment]
As there isn't any feedback I merge this now - if you have any comments, simply create follow-up PRs on it - thanks!

File: features/automation/ruledsl.md [hunk 1]
```diff
@@ -6,4 +6,273 @@ layout: documentation
...
+    * [Rule Triggers](#rule-triggers)
+    * [Event-based Triggers](#event-based-triggers)
+    * [Time-based Triggers](#time-based-triggers)
+    * [System-based Triggers](#system-based-triggers)
+  * [Scripts](#scripts)
+    * [Implicit Variables inside the Execution Block](#implicit-variables-inside-the-execution-block)
+    * [Concurrency Guard](#concurrency-guard)
+    * [Logging](#logging)
+  * [Example](#example)
+  * [Differences to openHAB 1](#diff)
```
Comment (maggu2810) [inline features/automation/ruledsl.md]: I believe that the diff section is missing.

**Comment by maggu2810** [comment]
Isn't the "ruledsl" better placed in the ESH docu and copied at build time=

**Pull request merged**